### PR TITLE
[bug][lixi] bug lixi random distribution #807

### PR DESCRIPTION
**Pull request by phongvu2812**
# Ticket
 #807

# General
- [x] No unnecessary files, debug logs, or commented code included.
- [ ] PR has at least 1 peer approval before senior review.
- [x] I fully understand the code I submitted.
- [ ] No hardcoded configuration or secrets.
- [ ] **Environment Variables** changes? (Leave unchecked if N/A)

# Design & Solution Clarity
- [ ] A design diagram (sequence / flow) is attached or linked.
- [ ] The implementation matches the design.
- [ ] All major flows and decision points are represented in the diagram.
- [ ] Error handling and edge cases are clearly defined in the design.

Design link (diagram / doc):
`<paste link here>`

# Testing
- [ ] Unit tests cover main logic and important edge cases.

> Note: Complex logic must not rely on manual testing only.

# UI evidence (For frontend only)
`<paste screenshots here>`

# Peer Review Checklist
- [ ] I understand the overall solution and flow.
- [ ] The code follows the provided design diagram.
- [ ] No obvious logic or security issues found.
- [ ] Tests are sufficient for the scope of this change.

### CODE REVIEW

**Review by TrongNgoVan — APPROVED** [review]

File: dong-service/utils/red_envelope_generator.go [hunk 1]
```diff
@@ -53,29 +54,42 @@ func internalGenerate(totalAmount, minAmount, maxAmount int64, totalClaims int)
...
 	const threshold int64 = 100000
 
-	if totalAmount < threshold {
-		amounts := internalGenerate(totalAmount, minAmount, maxAmount, totalClaims)
+	scaledMin := (minAmount + roundingUnit - 1) / roundingUnit
+	scaledMax := maxAmount / roundingUnit
+	scaledTotal := totalAmount / roundingUnit
+	remainder := totalAmount % roundingUnit
 
+	if totalAmount < threshold || scaledMin >= scaledMax || (scaledMax*roundingUnit)+remainder > maxAmount {
```
Comment (hoangdohuy) [inline dong-service/utils/red_envelope_generator.go]: không cần kiểm soát chẵn tiền nữa, bỏ ra cho hàm random tùy ý chia luôn
